### PR TITLE
Backport LoggerFactory to 1.x series 

### PR DESCRIPTION
**Pull request by FunFunFine**
See #655

This MR backports #629 to Cats-Effect 2 compatible release series.

### CODE REVIEW

**Comment by rossabaker** [comment]
This seems good to me, but I don't see that CI ran, nor do I see a place to approve a run for a first-time contributor. 🤔

**Comment by rossabaker** [comment]
I added my own commit, and that triggered something that will check the rest.

**Comment by rossabaker** [comment]
Okay, looks like we've got binary compatibility issues.  Seems like we would have hard to sort these out in 2.x.  I don't have time to chase it, but happy to review it if @FunFunFine or someone else can green it up.

File: core/shared/src/main/scala/org/typelevel/log4cats/LoggerFactory.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("""
+Implicit not found for LoggerFactory[${F}].
+Information can be found here: https://log4cats.github.io/logging-capability.html
```
Comment (danicheg) [inline core/shared/src/main/scala/org/typelevel/log4cats/LoggerFactory.scala:23]: Here is the dead link. We can pick up the fix from the main.
```suggestion
Learn about LoggerFactory at https://typelevel.org/log4cats/#logging-using-capabilities
```

**Comment by danicheg** [comment]
@FunFunFine Anton, let me know if you need help with Mima's issues.

**Comment by danicheg** [comment]
I think I beat the final boss. @rossabaker you have done huge work on reviewing this. It'd be cool if you do it once again.

**Comment by FunFunFine** [comment]
Thank you guys for taking this, I had quite a tough couple of weeks at work, so you really helped me (and library users).

**Comment by rossabaker** [comment]
This is being released as 1.7.0.

**Pull request merged**